### PR TITLE
media_folder on widged level

### PR DESCRIPTION
**Pull request by vencax**
Some of webs has divided uploads into separate folders so the media_folder settings needs to be on widget level. This PR solves this without change of original behaviour.
It might be useful for others as well.
Cheers :)

### CODE REVIEW

File: src/valueObjects/AssetProxy.js [hunk 1]
```diff
@@ -8,14 +8,16 @@ export const setStore = (storeObj) => {
   store = storeObj;
 };
 
-export default function AssetProxy(value, fileObj, uploaded = false) {
+export default function AssetProxy(value, fileObj, uploaded = false, field_config = null) {
   const config = store.getState().config;
+  const media_folder = field_config && field_config.has('media_folder') ? field_config.get('media_folder') : config.get('media_folder');
+  const public_folder = field_config && field_config.has('public_folder') ? field_config.get('public_folder') : '/' + media_folder;
```
Comment (vencax) [inline src/valueObjects/AssetProxy.js]: I cannot see if your version has any sideeffects. I prefer longer but self-explaining version

**Comment by vencax** [comment]
Ok, changed ... cheers :)

File: src/valueObjects/AssetProxy.js [hunk 2]
```diff
@@ -8,14 +8,16 @@ export const setStore = (storeObj) => {
   store = storeObj;
 };
 
-export default function AssetProxy(value, fileObj, uploaded = false) {
+export default function AssetProxy(value, fileObj, uploaded = false, field = null) {
   const config = store.getState().config;
+  const media_folder = (field && field.has('media_folder')) ? field.get('media_folder') : config.get('media_folder');
+  const public_folder = (field && field.has('public_folder')) ? field.get('public_folder') : '/' + media_folder;
```
Comment (erquhart) [inline src/valueObjects/AssetProxy.js]: `public_folder` now falls back to `'/' + media_folder` instead of `config.get('public_folder')`?

**Comment by erquhart** [comment]
@vencax I pushed some changes to avoid having the asset proxy worry about the concept of a field, and some docs tweaks.

Can you test it against your use case and confirm whether everything is working?

**Comment by vencax** [comment]
Some fixes due to copypaste was needed ... :)

**Comment by erquhart** [comment]
Ah! Great catches, thank you for that.

**Comment by erquhart** [comment]
@vencax I'm now thinking we should hold off on this. #350 is a big priority, and allowing multiple media directories becomes much more complex when the Media Library UI is in place.

I'm writing up plans now, but the priority for this goes:

1. Provide a media library UI that opens when you click an image/file field
2. Allow multiple asset collections (along with allowing a field to be restricted to a specific asset collection) - this will do what you need
3. Provide a "front page" UI for asset collections, outside of the pop up UI

Apologies that I didn't already bring this up, didn't consider that there would be a conflict. Please chime in on #350 if you have thoughts on it.

**Comment by vencax** [comment]
ok, i have what I need. It works. You have it in this PR. Do what you consider best for your comunity ...

**Comment by erquhart** [comment]
@vencax after #350, the goal of your PR would be addressed by allowing asset collections to be defined in the config, the same way that we have content collections currently. You would also be able to lock a widget to one or more specific asset collections, similar to what you're doing in this PR, except by asset collection name.

**Comment by ilyabo** [comment]
@erquhart I also really need this for making Netlify CMS play with Gatsby nicely. Since an implementation for #350 has been merged, is it now possible to have multiple asset collections and lock a widget to one asset collection as you describe? I tried to find some documentation about this, but couldn't.

**Pull request closed without merging**